src/scipp/coords/transform_coords.py

```diff
@@ -204,12 +204,26 @@ def _transform_dataset(original: Dataset, targets: Set[str], graph: Graph, *,
     # since different items may have different attributes. Unless we have
     # clear performance requirements we therefore go with the safe and
     # simple solution
-    return Dataset(
-        data={
-            name: _transform_data_array(
-                original[name], targets=targets, graph=graph, options=options)
-            for name in original
-        })
+    if len(original) > 0:
+        return Dataset(
+            data={
+                name: _transform_data_array(
+                    original[name], targets=targets, graph=graph, options=options)
+                for name in original
+            })
+
+    # Cannot keep attributes in output anyway.
+    # So make sure they are removed as early as possible.
```

Comment:
Maybe a sign that the coord->attr mapping mechanism is flawed (in general, not specific to `transform_coords`)?